pipelines/fault_recovery.py

```
@@ -0,0 +1,1797 @@
+        if isinstance(item.get("quantity"), str) and oc.is_true(item.get("measured"))
+    }
+    missing = sorted(
+        quantity
+        for quantity, target in expected.items()
+        if target is not None and quantity not in recorded_quantities
+    )
+    if missing:
+        return [
+            f"{where}.result: OUTCOME_NOT_REPRODUCIBLE — the replay derives "
+            f"measurements {missing} that the record does not carry as "
+            "measured readings"
```

Ruff (0.16.3)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)